2-mandatory/1-create-functions.js

```diff
@@ -3,15 +3,31 @@ Write a function that:
 - Accepts an array as a parameter.
 - Returns a new array containing the first five elements of the passed array.
 */
-function first5() {
+const arr = [1, 9, 7, 8, 4, 2, 8];
+function first5(arr) {
+  return arr.slice(0, 5);
 }
-
+// console.log(first5(arr));
 /*
 Write a function that:
 - Accepts an array as a parameter.
 - Returns a new array containing the same elements, except sorted.
 */
-function sortArray() {
+const createDeepCopy = (input) => {
+  if (typeof input !== "object") {
+    return input;
+  }
+  let copy = Array.isArray(input) ? [] : {};
+  for (let key in input) {
+    const value = input[key];
+    copy[key] = createDeepCopy(value);
+  }
```

Comment:
The question asks you to accept an array, so you don't have to deal with a map -  you could just return an error if the parameter isn't an array. Then you could use Array.from(array) to make a copy by value rather than reference. But it is good that you clearly understand how to make a non-reference copy for both and array and a map.